lib/fluent/plugin/parser_winevt_xml.rb

```diff
@@ -0,0 +1,33 @@
+require 'fluent/plugin/parser'
+require 'nokogiri'
+
+module Fluent::Plugin
+  class WinevtXMLparser < Parser
+    Fluent::Plugin.register_parser('winevt_xml', self)
+
+    def parse(text)
+      record = {}
+      doc = Nokogiri::XML(text)
+      record["ProviderName"]          = (doc/'Event'/'System'/"Provider").attribute("Name").text rescue nil
```

Comment:
Is call `doc/'Event'/'System'` for each element slow?
How about `system_elem = doc/'Event'/'System'`?

Reply:
Oops. I've fixed in https://github.com/fluent/fluent-plugin-windows-eventlog/pull/21/commits/2ad9629be9f6c5858fa5265b19643e8b43558fe1.